File: easy/242./242.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Boolean result = true;

        if(s.length() != t.length()) { 
            return false;
        }

        HashMap<Character, Integer> mapS = new HashMap<>();
        HashMap<Character, Integer> mapT = new HashMap<>();

        for(int i = 0; i <= s.length() - 1; i++) {
           if(mapS.containsKey(s.charAt(i))) {
                mapS.put(s.charAt(i), mapS.get(s.charAt(i))+1);
            } else {
                mapS.put(s.charAt(i), 1);
            } 
        }

        for(int i = 0; i <= t.length() - 1; i++) {
            if(mapT.containsKey(t.charAt(i))) {
                mapT.put(t.charAt(i), mapT.get(t.charAt(i))+1);
            } else {
                mapT.put(t.charAt(i), 1);
            }
        }

        for(Character c : mapS.keySet()) {
            if(!mapS.get(c).equals(mapT.get(c))) {
                result = false;
            }
        }

        return result;
    }
}

File: easy/242./242.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length) {
            return false;
        }

        Dictionary<char, int> mapStringS = new Dictionary<char, int>();
        Dictionary<char, int> mapStringT = new Dictionary<char, int>();

        foreach(char c in s) {
            if(mapStringS.ContainsKey(c)) {
                mapStringS[c] += 1;
            } else {
                mapStringS.Add(c, 1);
            }
        }

        foreach(char c in t) {
            if(mapStringT.ContainsKey(c)) {
                mapStringT[c] += 1;
            } else {
                mapStringT.Add(c, 1);
            }
        }

        foreach(var entry in mapStringS.Keys) {
            if(!(mapStringT.TryGetValue(entry, out var value) && mapStringS[entry].Equals(value))) {
                return false;
            }
        }

        return true;
    }
}